src/ol_infrastructure/applications/airbyte/__main__.py

```diff
@@ -0,0 +1,268 @@
+    assume_role_policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Principal": {"Service": "ec2.amazonaws.com"},
+            },
+        }
+    ),
+    path="/ol-infrastructure/airbyte-server/role/",
+    tags=aws_config.tags,
+)
+iam.RolePolicyAttachment(
+    f"airbyte-server-describe-instance-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["describe_instances"],
+    role=airbyte_server_instance_role.name,
+)
+iam.RolePolicyAttachment(
+    f"airbyte-server-route53-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["route53_odl_zone_records"],
+    role=airbyte_server_instance_role.name,
+)
+airbyte_server_instance_profile = iam.InstanceProfile(
+    f"airbyte-server-instance-profile-{env_name}",
+    role=airbyte_server_instance_role.name,
+    path="/ol-infrastructure/airbyte-server/profile/",
+)
+
+# Vault policy definition
+airbyte_server_vault_policy = vault.Policy(
+    "airbyte-server-vault-policy",
+    name="airbyte-server",
+    policy=Path(__file__).parent.joinpath("airbyte_server_policy.hcl").read_text(),
+)
+# Register Tika AMI for Vault AWS auth
+vault.aws.AuthBackendRole(
+    "airbyte-server-ami-ec2-vault-auth",
+    backend="aws",
+    auth_type="iam",
+    role="airbyte-server",
+    inferred_entity_type="ec2_instance",
+    inferred_aws_region=aws_config.region,
+    bound_iam_instance_profile_arns=[airbyte_server_instance_profile.arn],
+    bound_ami_ids=[airbyte_server_ami.id],
+    bound_account_ids=[aws_account.account_id],
+    bound_vpc_ids=[vpc_id],
+    token_policies=[airbyte_server_vault_policy.name],
+)
+
+##################################
+#     Network Access Control     #
+##################################
+# Create security group
+airbyte_server_security_group = ec2.SecurityGroup(
+    f"airbyte-server-security-group-{env_name}",
+    name=f"airbyte-server-operations-{env_name}",
+    description="Access control for airbyte servers",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            protocol="tcp",
+            from_port=DEFAULT_HTTPS_PORT,
+            to_port=DEFAULT_HTTPS_PORT,
+            cidr_blocks=["0.0.0.0/0"],
+            description=f"Allow traffic to the airbyte server on port {DEFAULT_HTTPS_PORT}",
+        ),
+    ],
+    egress=default_egress_args,
+    vpc_id=vpc_id,
+)
+
+###################################
+#     Web Node EC2 Deployment     #
+###################################
+lb_config = OLLoadBalancerConfig(
+    subnets=target_vpc["subnet_ids"],
+    security_groups=[airbyte_server_security_group],
+    tags=aws_config.merged_tags({"Name": airbyte_server_tag}),
+)
+
+tg_config = OLTargetGroupConfig(
+    vpc_id=vpc_id,
+    health_check_interval=60,
+    health_check_matcher="200-399",
+    health_check_path="/.pomerium",
```

Comment:
Is there an Airbyte specific path that we can use for the healthcheck to know if the underlying service is happy?

Reply:
Not that pomerium is going to let us talk to without authenticating. We will need to do nginx/caddy in front to sneak around for specific paths. Probably a round 2 worthy task.

Reply:
Moved this item to the phase two issue in the data milestone https://github.com/mitodl/ol-infrastructure/issues/899